src/module.c

```diff
@@ -464,6 +465,149 @@ int GetSingleDocumentCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
   return REDISMODULE_OK;
 }
 
+int DictDumpCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  char *err = NULL;
+  if (argc != 2) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  const char *dictName = RedisModule_StringPtrLen(argv[1], NULL);
+
+  char *error;
+  int retVal = SpellCheck_DictDump(ctx, dictName, &error);
+  if (retVal < 0) {
+    RedisModule_ReplyWithError(ctx, error);
+  }
+
+  return REDISMODULE_OK;
+}
+
+int DictDelCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  char *err = NULL;
+  if (argc < 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  const char *dictName = RedisModule_StringPtrLen(argv[1], NULL);
+
+  char *error;
+  int retVal = SpellCheck_DictDel(ctx, dictName, argv + 2, argc - 2, &error);
+  if (retVal < 0) {
+    RedisModule_ReplyWithError(ctx, error);
+  } else {
+    RedisModule_ReplyWithLongLong(ctx, retVal);
+  }
+
+  return REDISMODULE_OK;
+}
+
+int DictAddCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  char *err = NULL;
+  if (argc < 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+
+  const char *dictName = RedisModule_StringPtrLen(argv[1], NULL);
+
+  char *error;
+  int retVal = SpellCheck_DictAdd(ctx, dictName, argv + 2, argc - 2, &error);
+  if (retVal < 0) {
+    RedisModule_ReplyWithError(ctx, error);
+  } else {
+    RedisModule_ReplyWithLongLong(ctx, retVal);
+  }
+
+  return REDISMODULE_OK;
+}
+
+int SpellCheckCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
```

Comment:
Should be moved into its own file.

Reply:
It uses lots of function that do not related to spell check that is why I thought its better to put it here